samples-generator/bin/3d-tiles-samples-generator.js

```diff
@@ -7,6 +7,9 @@ var gltfPipeline = require('gltf-pipeline');
 var path = require('path');
```

Comment:
Legacy pnts contain extraneous attributes in their batch table json. Only `byteOffset`, `componentType` and `type` are allowed for binary properties. See https://github.com/CesiumGS/3d-tiles/tree/master/specification/TileFormats/BatchTable#implementation-example

From `pointCloudWithPerPointProperties.pnts`
```
{
  "temperature": {
    "name": "temperature",
    "count": 1000,
    "byteLength": 4000,
    "byteOffset": 0,
    "componentType": "FLOAT",
    "type": "SCALAR",
    "min": [
      0
    ],
    "max": [
      0.779883630837293
    ]
  },
  "secondaryColor": {
    "name": "secondaryColor",
    "count": 1000,
    "byteLength": 12000,
    "byteOffset": 4000,
    "componentType": "FLOAT",
    "type": "VEC3",
    "min": [
      0.00007369695231318474,
      0,
      0
    ],
    "max": [
      0.9992079983931035,
      0,
      0
    ]
  }
}
```

From `pointCloudBatched.pnts`:

```
{
  "name": [
    "section0",
    "section1",
    "section2",
    "section3",
    "section4",
    "section5",
    "section6",
    "section7"
  ],
  "dimensions": {
    "name": "dimensions",
    "count": 8,
    "byteOffset": 0,
    "byteLength": 96,
    "componentType": "FLOAT",
    "type": "VEC3",
    "min": [
      0.07815812015905976,
      0.07774908747524023,
      0.05862906435504556
    ],
    "max": [
      0.851557791698724,
      0.7785842611920089,
      0.9744280448649079
    ]
  },
  "id": {
    "name": "id",
    "count": 8,
    "componentType": "UNSIGNED_INT",
    "type": "SCALAR",
    "byteOffset": {
      "type": "Buffer",
      "data": [
        228,
        242,
        211,
        62,
        86,
        70,
        38,
        63,
        170,
        46,
        33,
        63,
        69,
        159,
        197,
        61,
        76,
        81,
        71,
        63,
        30,
        116,
        121,
        63,
        177,
        255,
        89,
        63,
        172,
        159,
        209,
        62,
        108,
        0,
        81,
        63,
        44,
        134,
        11,
        63,
        29,
        16,
        42,
        62,
        47,
        253,
        0,
        63,
        86,
        17,
        84,
        63,
        204,
        206,
        34,
        63,
        8,
        37,
        112,
        61,
        93,
        17,
        160,
        61,
        139,
        249,
        76,
        62,
        157,
        44,
        184,
        61,
        32,
        120,
        31,
        63,
        234,
        58,
        159,
        61,
        235,
        227,
        234,
        61,
        184,
        123,
        60,
        62,
        253,
        116,
        26,
        63,
        164,
        47,
        206,
        62
      ]
    },
    "byteLength": 32,
    "min": [
      0
    ],
    "max": [
      7
    ]
  }
}
```

Reply:
It might be a good idea to diff `.pnts` files between this branch and master to make sure nothing changed about the legacy path.